xsuite/src/proxy/csproxy.ts

```diff
@@ -0,0 +1,131 @@
+import { addressLikeToBechAddress } from "../data/addressLike";
+import { e, eCodeMetadata, EncodableAccount } from "../data/encoding";
+import { Kvs } from "../data/kvs";
+import { BroadTx, Proxy } from "./proxy";
+
+export class CSProxy extends Proxy {
+  autoGenerateBlocks: boolean;
+  waitCompletedTimeout: number;
+
+  constructor(params: CSProxyParams) {
+    super(params);
+
+    this.autoGenerateBlocks = params.autoGenerateBlocks;
+    this.waitCompletedTimeout = params.waitCompletedTimeout ?? 250;
+  }
+
+  async setAccount(account: EncodableAccount) {
+    const previousKvs = await this.getAccountKvs(account.address);
+    const newAccount = accountToRawAccount(account, previousKvs);
+
+    const result = this.fetch("/simulator/set-state", [newAccount]);
+
+    if (this.autoGenerateBlocks) {
```

Comment:
This logic should be in the world. The purpose of CSProxy should just be to simplify fetching the chain simulator, not to have chain logic.

Reply:
Even though this seems to be easier to do that in the other places, I still stand by my argument that there is no problem in having chain logic here.